src/main/java/network/radicle/jetbrains/radiclejetbrainsplugin/config/RadicleSeedNodeDecorator.java

```diff
@@ -0,0 +1,147 @@
+package network.radicle.jetbrains.radiclejetbrainsplugin.config;
+
+import com.intellij.openapi.actionSystem.ActionToolbarPosition;
+import com.intellij.ui.AnActionButton;
+import com.intellij.ui.AnActionButtonRunnable;
+import com.intellij.ui.ToolbarDecorator;
+import com.intellij.ui.table.JBTable;
+import network.radicle.jetbrains.radiclejetbrainsplugin.RadicleBundle;
+import network.radicle.jetbrains.radiclejetbrainsplugin.dialog.AddSeedNodeDialog;
+import network.radicle.jetbrains.radiclejetbrainsplugin.dialog.ConfirmationDialog;
+
+import javax.swing.*;
+import javax.swing.table.DefaultTableModel;
+import java.util.ArrayList;
+import java.util.List;
+
+public class RadicleSeedNodeDecorator {
+
+    private final RadicleSettingsHandler radicleSettingsHandler;
+    private RadicleSettings settings;
+    protected JBTable table;
+    private List<String> loadedSeedNodes;
+    private ArrayList<String> cpLoadedSeedNodes;
```

Comment:
please use the most abstract interface semantics possible, rather than concrete classes (e.g. `List` or even `Collection`)